lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -679,9 +747,7 @@ PromotedParamCloner::populateCloned() {
 
       NewPromotedArgs[ArgNo] = promotedArg;
 
-      // All uses of the promoted box should either be projections, which are
-      // folded when visited, or copy/destroy operations which are ignored.
-      entryArgs.push_back(SILValue());
+      entryArgs.push_back(promotedArg);
```

Comment:
Why is this being changed here? I need to understand that.

Reply:
I reverted this change. Thanks for pointing out. Not mapping entryArg to promotedArg is good here, if we were to have new kind of users for promoted boxes without implementing the visit* functions in PromotedParamCloner, SILCLoner will then assert saying there is no mapped value.